ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -237,7 +241,17 @@ const ConfirmPageContainer = (props) => {
                       <Button
                         type="inline"
                         className="confirm-page-container-content__link"
-                        onClick={() => setShowDepositPopover(true)}
+                        onClick={() => {
+                          openBuyCryptoInPdapp();
+                          trackEvent({
+                            event: EVENT_NAMES.NAV_BUY_BUTTON_CLICKED,
+                            category: EVENT.CATEGORIES.NAVIGATION,
+                            properties: {
+                              location: 'Transaction Confirmation',
+                              text: 'Buy',
+                            },
+                          });
+                        }}
```

Comment:
We try to keep this component for layout purpose only, any function invocation can be written [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js) and passed as prop to this component.

Reply:
@jpuri The base component is a class component; knowing that I have created a custom hook to centralize the logic of redirecting to the Buy Crypto in portfolio dapp, how can we use this hook within a class component?

Reply:
@jpuri this was the main reason I used the hook directly within `confirm-page-container.component.js` component, one example for a similar implementation is the usage of `useTransactionInsights`.

Reply:
Make sense, can you plz add a comment about this.

We plan to make `confirm-transaction-base` also functional and once that is done we can extract this out.

Reply:
I have addressed your feedback here https://github.com/MetaMask/metamask-extension/pull/17689/commits/fdb3df73299cedb243379336039b321186ce029a